### PR TITLE
chore: slither annotations

### PR DESCRIPTION
**Pull request by vzctl**
Add annotations to disable Slither false-positives and enable the Slither check to fail on new errors.

`TODO` items require extra input/fixes.

### CODE REVIEW

File: contracts/JetStakingV1.sol [hunk 1]
```diff
@@ -818,6 +821,8 @@ contract JetStakingV1 is AdminControlled {
         returns (uint256)
     {
         require(lastUpdate <= block.timestamp, "INVALID_LAST_UPDATE");
+        // TODO the line below is never executed because of the require() above
```
Comment (apotheon) [inline contracts/JetStakingV1.sol:824]: This TODO is inaccurate and should be removed.

File: contracts/JetStakingV1.sol [hunk 2]
```diff
@@ -911,6 +916,7 @@ contract JetStakingV1 is AdminControlled {
     /// @dev gets the total amount of staked aurora
     /// @return totalAmountOfStakedAurora + latest reward schedule
     function getTotalAmountOfStakedAurora() external view returns (uint256) {
+        // slither-disable-next-line incorrect-equality
         if (touchedAt == 0) return 0;
```
Comment (apotheon) [inline contracts/JetStakingV1.sol:920]: Should this read `if (touchedAt < 1)` ?

**Comment by vzctl** [comment]
I think it is ready to be merged! @0x3bfc if the list of TODOs is fine with you - we can create issues and address them separately!

File: contracts/AdminControlled.sol [hunk 3]
```diff
@@ -93,13 +93,15 @@ contract AdminControlled is UUPSUpgradeable, AccessControlUpgradeable {
     /// @param target the target contract address
     /// @param data is the ABI encoded function signature and its values.
     /// @custom:oz-upgrades-unsafe-allow delegatecall
+    // slither-disable-next-line low-level-calls controlled-delegatecall
```
Comment (0x3bfc) [inline contracts/AdminControlled.sol:96]: Do we need this line ?

File: contracts/JetStakingV1.sol [hunk 4]
```diff
@@ -356,6 +356,8 @@ contract JetStakingV1 is AdminControlled {
             stream.auroraDepositAmount = newAuroraDepositAmount;
             // update stream reward schedules
             _updateStreamRewardSchedules(streamId, rewardTokenAmount);
+            // TODO reorder the events
```
Comment (0x3bfc) [inline contracts/JetStakingV1.sol:359]: Please reference this as an issue.

File: contracts/JetStakingV1.sol [hunk 5]
```diff
@@ -944,7 +949,7 @@ contract JetStakingV1 is AdminControlled {
                 break;
             }
         }
-        // find end index
+        // slither-disable-next-line incorrect-equality
```
Comment (0x3bfc) [inline contracts/JetStakingV1.sol:952]: add the missing comment `// find end index`

File: contracts/JetStakingV1.sol [hunk 6]
```diff
@@ -1058,6 +1067,9 @@ contract JetStakingV1 is AdminControlled {
         uint256 reward = ((streams[streamId].rps -
             userAccount.rpsDuringLastClaim[streamId]) *
             userAccount.streamShares) / RPS_MULTIPLIER;
+
+        // TODO replace with reward < 1 instead?
```
Comment (0x3bfc) [inline contracts/JetStakingV1.sol:1071]: Reference this an issue `// TODO replace with reward < 1 instead?`

**Review by 0x3bfc — APPROVED** [review]

**Pull request merged**